### PR TITLE
Allow GIFs missing both Global and Local Colormaps to still decode

### PR DESCRIPTION
**Pull request by az6667**
Ensure that a GIF without any Global or Local color tables is still
decoded by libgd.

GIF89a spec indicates conforming image files need not have
Global or Local color tables at all.

Spec recommends creating custom color map in that situation, and
that at least Black+White as first two entries, to ensure B&W images
are decoded.

Some commonly used single-pixel GIFs found around the web are
undecoded by libgd otherwise. Test case has been included.

References:
https://www.w3.org/Graphics/GIF/spec-gif89a.txt
http://probablyprogramming.com/2009/03/15/the-tiniest-gif-ever

### CODE REVIEW

File: tests/gif/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ LIST(APPEND TESTS_FILES
 	bug00181
 	bug00227
 	gif_null
+	git_nocolormaps
```
Comment (cmb69) [inline tests/gif/CMakeLists.txt]: Typo: should be `gif_nocolormaps`.

Comment (az6667) [reply tests/gif/CMakeLists.txt]: well that's embarrasing. thankyou.

File: src/gd_gif_in.c [hunk 2]
```diff
@@ -315,8 +315,9 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFromGifCtx(gdIOCtxPtr fd)
 			ReadImage(im, fd, width, height, localColorMap, BitSet(buf[8], INTERLACE), &ZeroDataBlock);
 		} else {
 			if(!haveGlobalColormap) {
-				gdImageDestroy(im);
-				return 0;
+				ColorMap[CM_RED][1] = 0xff;	//Still a valid gif, apply simple default palette as per spec
```
Comment (vapier) [inline src/gd_gif_in.c]: can you move the comment above the line instead of being inline ?

also put a space after the `//`

Comment (az6667) [reply src/gd_gif_in.c]: yup sure

**Comment by vapier** [comment]
thanks!

**Pull request merged**